Framework/PythonInterface/mantid/plots/plotfunctions.py

```diff
@@ -217,8 +218,10 @@ def plot(workspaces, spectrum_nums=None, wksp_indices=None, errors=False,
     if superplot and not waterfall and not tiled:
         fig.canvas.manager.superplot_toggle()
         if not spectrum_nums and not wksp_indices:
-            workspace_names = [ws.getName() for ws in workspaces]
+            workspace_names = [ws.name() for ws in workspaces]
             fig.canvas.manager.superplot.set_workspaces(workspace_names)
+            fig.canvas.manager.superplot.set_bin_mode(plot_kwargs and "axis" in plot_kwargs and plot_kwargs["axis"] == MantidAxType.BIN)
```

Comment:
Can we add a test to `plotfunctionTest.py` to ensure `set_bin_mode` is called correctly for both the case where `MantixAxType.BIN` is passed in the nested dict, and that it gracefully handles cases where it we don't.

Currently, this looks like it handles all edge cases. But because of the complexity of accessing a nested item if someone does some overzealous refactoring they might break this subtly in the future without us noticing easily.

Reply:
I added a test.